pkg/apiserver/controller/fs/fs.go

```diff
@@ -154,21 +150,54 @@ func (s *FileSystemService) GetFileSystem(fsID string) (models.FileSystem, error
 
 // DeleteFileSystem the function which performs the operation of delete file system
 func (s *FileSystemService) DeleteFileSystem(ctx *logger.RequestContext, fsID string) error {
-	err := models.DeleteFileSystem(fsID)
+	return models.WithTransaction(database.DB, func(tx *gorm.DB) error {
+		if err := models.DeleteFileSystem(tx, fsID); err != nil {
+			ctx.Logging().Errorf("delete fs[%s] failed error[%v]", fsID, err)
+			ctx.ErrorCode = common.FileSystemDataBaseError
+			return err
+		}
+		// delete cache config if exist
+		if err := models.DeleteFSCacheConfig(tx, fsID); err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				return nil
+			}
+			ctx.Logging().Errorf("delete fs[%s] cache config failed error[%v]", fsID, err)
+			ctx.ErrorCode = common.FileSystemDataBaseError
+			return err
+		}
+		if err := DeletePvPvc(fsID); err != nil {
+			ctx.Logging().Errorf("delete deletePvPvc for fs[%s] err: %v", fsID, err)
+			return err
+		}
+		return nil
+	})
+}
+
+func DeletePvPvc(fsID string) error {
+	clusters, err := models.ListCluster(0, 0, nil, "")
 	if err != nil {
-		ctx.Logging().Errorf("delete fs[%s] failed error[%v]", fsID, err)
-		ctx.ErrorCode = common.FileSystemDataBaseError
-		return err
+		return fmt.Errorf("list clusters failed")
 	}
-	// delete cache config if exist
-	err = models.DeleteFSCacheConfig(ctx.Logging(), fsID)
-	if err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return nil
+	for _, cluster := range clusters {
+		switch cluster.ClusterType {
+		case schema.KubernetesType:
+			runtimeSvc, err := runtime.GetOrCreateRuntime(cluster)
+			if err != nil {
+				return err
+			}
+			k8sRuntime := runtimeSvc.(*runtime.KubeRuntime)
+			for _, ns := range cluster.NamespaceList {
+				// delete pvc manually. pv will be deleted automatically
+				err := k8sRuntime.DeletePersistentVolumeClaim(ns, schema.ConcatenatePVCName(fsID), metav1.DeleteOptions{})
+				if err != nil && !strings.Contains(err.Error(), "not found") {
```

Comment:
Kubernetes has errors package `k8s.io/apimachinery/pkg/api/errors`， `errors.IsNotFound(err)` might be better to this.